WowPacketParser/Parsing/Parsers/UpdateHandler.cs

```diff
@@ -172,175 +178,338 @@ public static void ReadObjectsBlock(Packet packet, object index)
+                            fieldType = typeof(AreaTriggerField);
                             break;
                         }
                         case ObjectType.SceneObject:
                         {
-                            key = UpdateFields.GetUpdateFieldName<SceneObjectField>(i);
+                            tuple = UpdateFields.GetUpdateFieldNameTuple<SceneObjectField>(i);
+                            fieldType = typeof(SceneObjectField);
                             break;
                         }
                         case ObjectType.Conversation:
                         {
-                            key = UpdateFields.GetUpdateFieldName<ConversationField>(i);
+                            tuple = UpdateFields.GetUpdateFieldNameTuple<ConversationField>(i);
+                            fieldType = typeof(ConversationField);
                             break;
                         }
                     }
                 }
-                // HACK...
-                if (ClientVersion.AddedInVersion(ClientVersionBuild.V6_2_3_20726))
+                int start = i;
+                if (tuple != null && tuple.Item1 != string.Empty)
                 {
-                    if (key == UnitField.UNIT_FIELD_FACTIONTEMPLATE.ToString())
-                        packet.AddValue(key, value + $" ({ StoreGetters.GetName(StoreNameType.Faction, (int)blockVal.UInt32Value, false) })", index);
-                    else
-                        packet.AddValue(key, value, index);
+                    key = tuple.Item1;
+                    size = tuple.Item2;
+                    start = tuple.Item3;
                 }
-                else
-                    packet.AddValue(key, value, index);
+                UpdateFieldType updateFieldType = UpdateFieldType.Default;
+
+                List<UpdateField> fieldData = new List<UpdateField>();
+                for (int k = start; k < i; ++k)
+                    fieldData.Add(new UpdateField(0));
+                fieldData.Add(blockVal);
+                int currentPosition = i + 1;
+                for (int k = i - start + 1; k < size; ++k)
+                {
+                    if (mask[currentPosition])
+                        fieldData.Add(new UpdateField(packet.ReadUInt32()));
+                    else
+                        fieldData.Add(new UpdateField(0));
 
-                dict.Add(i, blockVal);
-            }
+                    ++currentPosition;
+                }
 
-            objectEnd = UpdateFields.GetUpdateField(ObjectDynamicField.OBJECT_DYNAMIC_END);
-            if (ClientVersion.AddedInVersion(ClientVersionBuild.V5_0_4_16016))
-            {
-                maskSize = packet.ReadByte();
-                updateMask = new int[maskSize];
-                for (var i = 0; i < maskSize; i++)
-                    updateMask[i] = packet.ReadInt32();
+                var memInfo = fieldType.GetMember(key);
```

Comment:
It's actually not too bad for enums. You can just construct an array of MemberInfo for the given type and reuse it. I think this is fine though.